src/mumble/UserLocalNicknameDialog.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "UserLocalNicknameDialog.h"
+#include "ClientUser.h"
+#include "Database.h"
+#include "MainWindow.h"
+
+#include <QtGui/QCloseEvent>
+#include <QtWidgets/QPushButton>
+
+// We define a global macro called 'g'. This can lead to issues when included code uses 'g' as a type or parameter name
+// (like protobuf 3.7 does). As such, for now, we have to make this our last include.
+#include "Global.h"
+
+UserLocalNicknameDialog::UserLocalNicknameDialog(unsigned int sessionId,
+											 QMap< unsigned int, UserLocalNicknameDialog * > *qmUserNicknameTracker)
+	: QDialog(nullptr), m_clientSession(sessionId), m_qmUserNicknameTracker(qmUserNicknameTracker) {
+	setupUi(this);
+	qleUserLocalNickname->setAccessibleName(tr("User nickname"));
+
+	ClientUser *user = ClientUser::get(sessionId);
+	if (user) {
+		QString title = tr("Adjusting local nickname for %1").arg(user->qsName);
+		setWindowTitle(title);
+		qleUserLocalNickname->setText(user->getLocalNickname());
+		m_originalNickname = qleUserLocalNickname->text();
+	}

```

Comment:
I think ideally the dialog should immediately close itself if the user is not found. Otherwise I could imagine this leading to strange results in terms of UX (e.g. the window not having a title in these cases)